dask/dataframe/io.py

```diff
@@ -462,14 +463,18 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
                  "must preserve the order of its input")
 
     # handle lock default based on wether we're writing to a single entity
+    _actual_get = get or _globals.get('get') or get_sync
```

Comment:
How do you feel about the following policy?

If the user provides `get` explicitly, or has set something in `_globals`, then use that.
Otherwise, if we're in a sequential writing case, use `get_sync`.  Otherwise, use `DataFrame._get`.

Now, if we're using a scheduler known to not need locks (`get_sync`, `dask.multiprocessing.get`) then don't use locks.  Otherwise, use locks.

Avoiding locks is nice, but in my experience not as nice as going away from the default scheduler for the rest of the computation.  My two cents anyway.  Happy to be overruled here, especially if you've had experiences to the contrary.